tests/phpunit/api/v3/PaymentScheduleTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+use CRM_MembershipExtras_Test_Fabricator_MembershipType as MembershipTypeFabricator;
+
+/**
+ * Class PaymentScheduleTest
+ */
+class PaymentScheduleTest extends BaseHeadlessTest {
+
+  /**
+   * @throws CiviCRM_API3_Exception
+   */
+  public function testGetMonthlyRollingMembershipTypeSchedule() {
```

Comment:
I would focus my test on the stuff that the API itself is doing, which include : 

1- testing something other than these : 

```

    CRM_MembershipExtras_Service_MembershipInstalmentsSchedule::MONTHLY,
    CRM_MembershipExtras_Service_MembershipInstalmentsSchedule::QUARTERLY,
    CRM_MembershipExtras_Service_MembershipInstalmentsSchedule::ANNUALLY,
```

will result in throwing an exception

2- test the format of the output, for example, that it should be an array with X elements (which vary if it monthly or annual or ..etc). as well as if at least one instalment in the array contains the following keys : 
`instalment_no`, `instalment_date`, 'instalment_tax_amount', 'instalment_amount' ..etc. And hence not test the content but rather if these keys exsit.

Reply:
@omarabuhussein   
Point 1 is done, expected exception added. 
Point 2: I may disagree with this, the reason I would check the content because the API is also return with format for example, currency symbol, so I would like to assert if they are equals. If the fields does not exist, the test will fail anyway.